src/kboot.c

```diff
@@ -2052,6 +2113,9 @@ int kboot_prepare_dt(void *fdt)
         dt = NULL;
     }
 
+    /* Need to init ISP early to carve out heap */
+    isp_init();
```

Comment:
this either needs `if (fdt_get_alias(dt, "isp"))`  or `isp_init` should silently fail if the ADT misses isp/isp0